src/content/docs/guides/json/0-json_intro.mdx

```diff
@@ -1,142 +1,192 @@
+  {
+      json game_data;
+      game_data = json_from_file("game_data.json");
+      string game_title = json_read_string(game_data, "gameTitle");
+
+      write("Game Title: ");
+      write_line(game_title);
+
+      free_json(game_data);
+
+      return 0;
+  }
+  ```
+
+  </TabItem>
+  <TabItem label="Python">
+
+```python
+from splashkit import *
+
+def main():
+    
+    game_data = json_from_file("game_data.json")
+    game_title = json_read_string(game_data, "gameTitle")
+    
+    if not game_title:
+        print("Error retrieving gameTitle.")
+    else:
+        print("Game Title:", game_title)
+    
+     free_json(game_data)
+   
+
+if __name__ == "__main__":
+    main()
+    ```
+
+   </TabItem>
+
+  
+</Tabs>
+
+In this code example, we're first using [Json From File](/api/json/#json-from-file) to load a JSON object with the details from the `game_data.json` file.
+Then we are getting the value of the key that matches `gameTitle` using [Json Read String](/api/json/#json-read-string). We write this to the console and free the JSON object using [Free Json](/api/json/#free-json) as we exit. By freeing the JSON object we are deallocating any memory that has been assigned to it, preventing any memory-related errors such as a `Segmentation Fault`. We can build this program with the following command.
+
+<Tabs syncKey="code-language">
+  <TabItem label="C++">
+
+  ```shell
+  g++ program.cpp -l SplashKit -o json_program
+  ```
+
+  </TabItem>
+</Tabs>
+
+And run it with:
+
+<Tabs syncKey="code-language">
+  <TabItem label="C++">
+
+  ```shell
+  ./json_program
+  ```
+  
+  </TabItem>
+  <TabItem label="Python">
+
+  ```shell
+  skm python3 program.py
+  ```
+</TabItem>
+</Tabs>
+
+When we run this program it should output the following to the console:
+
+```plaintext
+Game Title: My New Game
+```
+
+### Checking Keys
+
+But what if we didn't have a `gameTitle` key in our JSON? Well, error messages will be produced indicating that this key is `null`. To prevent this, we can use the [Json Has Key](/api/json/#json-has-key) function to check if the key is present and then do actions based on whether it has been found or not. We could then turn the previous example into the following code:
+
+<Tabs syncKey="code-language">
+  <TabItem label="C++">
+
+  ```cpp
+  #include "splashkit.h"
+  int main()
+  {
+      json game_data;
+      game_data = json_from_file("game_data.json");
+
+      if(json_has_key(game_data, "gameTitle"))
+      {
+          string game_title = json_read_string(game_data, "gameTitle");
+          write("Game Title: ");
+          write_line(game_title);
+      }
+      else
+      {
+          write_line("Key \"gameTitle\" not found.");
+      }
+
+      free_json(game_data);
+      
+      return 0;
+  }
+  ```
+
+  </TabItem>
+  <TabItem label="Python">
+
+  ```python
+ from splashkit import *
+
+def main():
+    
+    game_data = json_from_file("game_data.json")
+     if json_has_key(game_data, "gameTitle"):

```

Comment:
This specific line causes an error as the indentation is incorrect.
This whole python code block needs to correct formatting of indentations.